appservice/src/intTest/groovy/nu/yona/server/RejectBuddyTest.groovy

```diff
@@ -1,154 +1,45 @@
 package nu.yona.server
 
 import groovy.json.*
-import spock.lang.Shared
 
-class RejectBuddyTest extends AbstractAppServiceIntegrationTest {
-
-	@Shared
-	def richardQuin
-	@Shared
-	def bobDunn
-	@Shared
-	def richardQuinBobBuddyURL
-	@Shared
-	def bobDunnRichardBuddyURL
-	@Shared
-	def bobDunnBuddyMessageRejectURL
-	@Shared
-	def bobDunnBuddyMessageProcessURL
-	@Shared
-	def richardQuinBuddyMessageRejectURL
-	@Shared
-	def richardQuinBuddyMessageProcessURL
-
-	def 'Add user Richard'(){
+class RejectBuddyTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Bob rejects Richard\'s buddy request"'()
+	{
 		given:
+			def richard = addRichard();
+			def bob = addBob();
+			appService.sendBuddyConnectRequest(richard, bob)
+			def rejectURL = appService.fetchBuddyConnectRequestMessage(bob).rejectURL
 
 		when:
-		richardQuin = addUser("Richard", "Quin")
+			def rejectMessage = "Sorry, not you"
+			def rejectResponse = appService.postMessageActionWithPassword(rejectURL, ["message" : rejectMessage], bob.password)
 
 		then:
-		richardQuin
-	}
+			rejectResponse.status == 200
 
-	def 'Add user Bob'(){
-		given:
+			// Verify connect message doesn't have actions anymore
+			def actionURLs = appService.fetchBuddyConnectRequestMessage(bob).rejectURL
+			!actionURLs?.size
 
-		when:
-		bobDunn = addUser("Bob", "Dunn")
-
-		then:
-		bobDunn
-	}
+			def processResult = appService.fetchBuddyConnectResponseMessage(richard)
+			processResult.message == rejectMessage
 
-	def 'Richard requests Bob to become his buddy'(){
-		given:
+			// Have the requesting user process the buddy connect response
+			def processResponse = appService.postMessageActionWithPassword(processResult.processURL, [ : ], richard.password)
+			processResponse.status == 200
 
-		when:
-		def response = appService.requestBuddy(richardQuin.url, """{
-				"_embedded":{
-					"user":{
-						"mobileNumber":"${bobDunn.mobileNumber}"
-					}
-				},
-				"message":"Would you like to be my buddy?",
-				"sendingStatus":"REQUESTED",
-				"receivingStatus":"REQUESTED"
-			}""", richardQuin.password)
-		if (response.status == 201) {
-			richardQuinBobBuddyURL = response.responseData._links.self.href
-		}
+			// Verify that Bob is not in Richard's buddy list anymore
+			def getRichardBudddiesResponse = appService.getBuddies(richard);
+			!getRichardBudddiesResponse.responseData?._embedded?.buddies
 
-		then:
-		response.status == 201
-		response.responseData._embedded.user.firstName == "Bob"
-		richardQuinBobBuddyURL.startsWith(richardQuin.url)
+			// Verify that Richard is not in Bob's buddy list anymore
+			def getBobBudddiesResponse = appService.getBuddies(bob);
+			!getBobBudddiesResponse.responseData?._embedded?.buddies
```

Comment:
In another test I saw just `def buddies = appService.getBuddies(bob); buddies.size() == 0`. That was easier to read.